### PR TITLE
[13.0][ADD] website_sale_coupon_page: New module to show the promotions on website

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT30853

Moved from https://github.com/OCA/e-commerce/pull/557 

please @joao-p-marques @victoralmau review it

### CODE REVIEW

**Review by victoralmau — APPROVED** [review]
Code review OK.

File: website_sale_coupon_page/README.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+Configuration
+=============
+
+To show the promotions/coupons on website you need to:
+
+#. Go to *Website > Products > Promotion Programs/Coupon Programs*.
+#. Select the program that we will publish.
+#. In the Website section, upload an image, add a text or both.
+#. Press on the smart button for publish the promotion/coupon.
```
Comment (pedrobaeza) [inline website_sale_coupon_page/README.rst]: After a preposition, always a gerund:

```suggestion
#. Press on the smart button for publishing the promotion/coupon.
```

File: website_sale_coupon_page/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+{
+    "name": "Website Sale Coupon Page",
+    "version": "13.0.1.0.0",
+    "category": "Website",
+    "website": "https://github.com/OCA/sale-promotion",
+    "author": "Tecnativa, Odoo Community Association (OCA)",
+    "license": "LGPL-3",
+    "application": False,
+    "installable": True,
+    "depends": ["website_sale_coupon"],
```
Comment (pedrobaeza) [inline website_sale_coupon_page/__manifest__.py:13]: I think you only need to depend on `website` and `sale_coupon`, not on this one.

File: website_sale_coupon_page/README.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+~~~~~~~
+
+* Tecnativa
+
+Contributors
+~~~~~~~~~~~~
+
+* `Tecnativa <https://www.tecnativa.com>`_:
+
+  * Carlos Roca
```
Comment (pedrobaeza) [inline website_sale_coupon_page/README.rst:77]: No credit for me as designer?

File: website_sale_coupon_page/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+                    "|",
+                    ("website_id", "=", False),
+                    ("website_id", "=", request.env.context.get("website_id")),
+                ]
+            )
+        )
+        promos = request.env["sale.coupon.program"]
+        for promo in all_promos:
+            if promo._is_valid_partner(request.env.user.partner_id):
+                promos |= promo
```
Comment (pedrobaeza) [inline website_sale_coupon_page/controllers/main.py]:
```suggestion
        promos = all_promos.filtered(lambda x: x._is_valid_partner(request.env.user.partner_id))
```

File: website_sale_coupon_page/security/ir.model.access.csv [hunk 5]
```diff
@@ -0,0 +1,2 @@
+id,name,model_id/id,group_id/id,perm_read,perm_write,perm_create,perm_unlink
+access_program_user,"Access user to programs",model_sale_coupon_program,,1,0,0,0
```
Comment (pedrobaeza) [inline website_sale_coupon_page/security/ir.model.access.csv]: Never give a permission without group. This is a total leak. Normal technique is to add 2 lines for public and portal users, but also record rules should be in place for restricting to published elements. Anyways, as the risk for leaking some data for promotions (like number of sales records, or the rules), we are not going to do this, and put `sudo` on the controller, so the only way to access this information is through that controller.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by CarlosRoca13** [comment]
@pedrobaeza changes done

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**